src/libraries/System.Reflection.MetadataLoadContext/src/System/Reflection/TypeLoading/Modules/Ecma/EcmaModule.TypeProvider.cs

```diff
@@ -48,14 +48,14 @@ public RoType GetGenericInstantiation(RoType genericType, ImmutableArray<RoType>
         public bool IsSystemType(RoType type) => type == Loader.TryGetCoreType(CoreType.Type);
         public PrimitiveTypeCode GetUnderlyingEnumType(RoType type) => type.GetEnumUnderlyingPrimitiveTypeCode(Loader);
 
-        public RoType GetTypeFromSerializedName(string name)
+        public RoType GetTypeFromSerializedName(string? name)
         {
             // Called when an attribute argument is of type System.Type ([MyAttribute(typeof(Foo))]
             // Parse an assembly-qualified name as Assembly.GetType() does. If the assembly part is missing, search in _module (the
             // module in which the custom attribute metadata was found.)
             if (name == null)
-                return null; // This gets hit if the custom attribute passes "(Type)null"
-            return Helpers.LoadTypeFromAssemblyQualifiedName(name, GetRoAssembly(), ignoreCase: false, throwOnError: true);
+                return null!; // This gets hit if the custom attribute passes "(Type)null"
```

Comment:
Intentionally forcing the return type as non-nullable here, as declaring it nullable would require marking the `ISignatureTypeProvider` and `ICustomAttributeTypeProvider` interfaces implemented by this class as having a nullable `RoType` type parameter. In turn this would result in pervasive and unnecessary nullability annotations in big parts of the codebase.